### PR TITLE
Refactor+Update RPC methods and Add Quill context provider

### LINKED ISSUE #173: Bug: Unknown response id errors when calling eth_accounts
Video explanation:
https://youtu.be/VI065g8rO1Q?t=486

This happens frequently (but not always) when calling `ethereum.request({ method: 'eth_accounts' })`.

E.g.
Screen Shot 2022-03-25 at 2 09 00 pm

### PR DESCRIPTION
**Pull request by voltrevo**
# Video demo & explanation (!)

Part 1: https://www.youtube.com/watch?v=75e3KscvHp4
Part 2: https://www.youtube.com/watch?v=M4ZsD-pA2c4

(Consider using 1.5x playback speed 😅 (ask me if you don't how to do this))

## What is this PR doing?

Fixes rpc - removes duplicate background script in quill page and uses window.ethereum to talk to the background the same(ish) way that pages do. Also adds type information and new rpc methods that replace the direct (and incorrect) controller accesses that were happening.

## How can these changes be manually tested?

Run the extension and check the existing functionality works and the bugs are no longer there.

## Does this PR resolve or contribute to any issues?

Resolves #173, #214, #215

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

File: extension/source/Controllers/QuillController.ts [hunk 1]
```diff
@@ -568,6 +566,72 @@ export default class QuillController extends BaseController<
...
+        if (origin === window.location.origin) {
+          return this.keyringController.state.wallets.map(
+            ({ address }) => address,
+          );
+        }
+
+        // Expose no accounts if this origin has not been approved, preventing
+        // account-requiring RPC methods from completing successfully
+        // only show address if account is unlocked
+        // FIXME: The comment above is not yet implemented.
```
Comment (jacque006) [inline extension/source/Controllers/QuillController.ts]: Would be good to link this to an issue. Also, given that this is not intended to be used in a production setting, do we need to add this limitation?

File: extension/source/QuillPage/QuillContext.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  if (!ctxVal) {
+    // This could be replaced by a nicer splash screen
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <QuillContext.Provider value={ctxVal}>{children}</QuillContext.Provider>
+  );
+}
```
Comment (jacque006) [inline extension/source/QuillPage/QuillContext.tsx:85]: @voltrevo This file and how it is consumed is the major refactor I did in this PR. I really didn't like delaying the ReactDOM init under an async function, so I consolidated that and other context initialization logic into this provider, turning it into more of a black box abstracting the complexity of that flow away.

File: extension/source/types/window.d.ts [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
+import { providers } from 'ethers';
+import { QuillInPageProvider } from '../PageContentScript/InPageProvider';
 
-export declare global {
+declare global {
   interface Window {
-    // TODO (merge-ok) Remove both of these when full controller linking is completed.
-    QuillController: () => QuillController;
-    KeyringController: () => KeyringController;
+    ethereum: QuillInPageProvider | providers.ExternalProvider | undefined;
```
Comment (jacque006) [inline extension/source/types/window.d.ts:6]: https://docs.ethers.io/v5/api/providers/other/#Web3Provider--ExternalProvider

**Review by jacque006 — APPROVED** [review]

**Comment by jacque006** [comment]
@kautukkundan Since I also made some changes to this, your review would be appreciated.

File: extension/source/QuillPage/QuillContext.tsx (hunk 2 above)
Comment (kautukkundan) [reply extension/source/QuillPage/QuillContext.tsx:85]: Thanks, this is useful.
Can definitely change "Loading..." with some static assets in the future.

**Review by kautukkundan — APPROVED** [review]
LGTM 🚢

File: extension/source/Controllers/QuillController.ts (hunk 1 above)
Comment (jacque006) [reply extension/source/Controllers/QuillController.ts]: https://github.com/web3well/bls-wallet/issues/224

**Pull request merged**